pandas/io/json/_json.py

```diff
@@ -302,6 +300,11 @@ def __init__(
             raise ValueError(msg)
 
         obj = obj.copy()
+        if obj.index.nlevels == 1 and isinstance(obj.index, MultiIndex):
+            obj.index = obj.index.get_level_values(0)
+
+        self.schema = build_table_schema(obj, index=self.index)
```

Comment:
can you add a comment here that we are now building the schema.

this is very confusing if you look at it because right after we are then modifying obj, so pls add a comment there as well: we need to convert the object to types we are actually serializing

Reply:
you can remove comment on L311

Reply:
> can you add a comment here that we are now building the schema.
What kind of comment do you mean ? Method name `build_table_schema` seems clear.

Reply:
I am also a little confused by this - so what happens if we have this single level MultiIndex but also a frame containing timedeltas. Does that still yield the expected behavior?

Reply:
Yes, in that case, generated json is the expected one and roundtrip remains impossible.
I will complete tests considering timedeltas.

Reply:
Does moving this to the end of the function change behavior at all? This still reads a little strange being here

Reply:
The purpose of moving it is to perform the schema creation _after_ the object copy. In my opinion it makes sense to do these operations once conditions raising an error have been passed, but I may me missing something. We can also let `build_table_schema` at the start then add the `obj.copy()` and the conversion into single index at the start too. It is possible and works.